src/lib/reasoners/shostak.ml

```diff
@@ -72,6 +89,7 @@ struct
         | Bitv t -> fprintf fmt "%a" BITV.print t
         | Adt t -> fprintf fmt "%a" ADT.print t
         | Term t -> fprintf fmt "%a" Expr.print t
+        | Abstract (k, t) -> fprintf fmt "@@%a(%a)" pp_abs_kind k CX.print t
```

Comment:
It does have a loose relationship with it since in both cases they are abstract names for constants, so I think that reusing `@` makes sense. Note that the abstracted semantic values are written `@kind(arg)` whereas the SMT-LIB abstract constants are written `@name` with no arguments.

I think this disambiguates them enough to make using `@` acceptable here, but I am OK with switching to another prefix (`%`?).

Reply:
In fact this printer is only used for debugging. We use the printer of `Expr` to print models and currently, this is the only printer we really need to keep SMT-LIB compliant. Let us keep `@` with parentheses.